mlir/lib/Dialect/SCF/Transforms/LoopPipelining.cpp

```diff
@@ -449,9 +449,9 @@ scf::ForOp LoopPipelinerInternal::createKernelLoop(
         arith::MulIOp::create(rewriter, loc, step, maxStageValue);
     newUb = arith::SubIOp::create(rewriter, loc, ub, maxStageByStep);
   }
-  auto newForOp =
-      scf::ForOp::create(rewriter, forOp.getLoc(), forOp.getLowerBound(), newUb,
-                         forOp.getStep(), newLoopArg);
+  auto newForOp = scf::ForOp::create(
```

Comment:
There is more in the logic that assumes signed comparison in the lb/ub calculation. I would fail the transformation for now.